lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -31,8 +31,46 @@ def colorized_message_lines(colorizer=::RSpec::Core::Formatters::ConsoleCodes)
           end
         end
 
-        def formatted_backtrace
-          backtrace_formatter.format_backtrace(exception_backtrace, example.metadata)
+        def formatted_backtrace(exception=@exception)
+          lines = backtrace_formatter.format_backtrace((exception.backtrace || []), example.metadata)
+          potential_lines = formatted_cause(exception)
+
+          lines += potential_lines unless potential_lines.empty?
+
+          lines
+        end
+
+        if RSpec::Support::RubyFeatures.supports_exception_cause?
+          def formatted_cause(exception)
+            exceptions = []
+
+            while exception.cause
+              exception = exception.cause
+              exceptions << exception
```

Comment:
We only want to generate the `formatted_backtrace` when an exception has a cause. We always pas in an exception so `unless exception.nil?` will return false and we will always end up generating a `formatted_backtrace`.  So I use the array to ensure that the original exception passed in did have a `cause`. It's not a neat solution but so far I have'nt been able to think of a way to refactor it.